examples/relative_requirements/requirements.txt

```diff
@@ -0,0 +1 @@
+./relative_package
```

Comment:
Just taking a quick scan over this PR. What's the use case here? Shouldn't something that exists at `./relative_package` just be a `py_library`? Or is true use case something like `<out of workspace root>/../relative_package`?

Reply:
The use case as I have it is that requirements.txt is a file with relative requirements that my IDE will use to resolve requirements without needing to know about Bazel py_library.

Additionally, if it’s not supported, we would need to add support for resolving a library’s dependencies from setup.py or the PEP equivalent, as my requirements.txt requiring a local library might in turn require additional dependencies from outside the workspace.

Reply:
The [original issue](https://github.com/bazelbuild/rules_python/issues/366) sums it up pretty well: basically vendored wheel files within a repo. It's unfortunate to have to use checked in wheel files, but it's sadly unavoidable for a couple of use cases where you don't want to setup an entire private package repo for a few KB of wheels. This PR makes them work the same as remote dependencies (i.e their transitive deps get pulled).

From my perspective, the support for relative non-wheel dependencies is just a helpful side-effect of the PR, it's not a use-case I expect to use.